### ISSUE TITLE
--version should also print the mir-json version

### ISSUE THREAD

**Issue by sauclovian-g** [GaloisInc/mir-json#77]
Currently if you ask the mir-json tools for their version they print the rustc version, which is important, but realistically they should also print the mir-json version. Maybe also the current JSON schema version too, now that we have one.

In order to do this we need a version, but there is actually one in the cargo metadata (maybe we should bump it for the recent JSON changes?) so this shouldn't be that difficult to arrange.

**Comment by RyanGlScott** [comment]
I agree that we should do this. I'm unclear if the `rustc` API offers a clean way to override the behavior of the `--version` flag, but if nothing else, we could reimplement `--version` ourselves. (The `rustc` implementation can be found [here](https://github.com/rust-lang/rust/blob/9110d925d0b3e4842376e830f4404a28e1aa2658/compiler/rustc_driver/src/lib.rs#L768-L788).)

**Comment by spernsteiner** [comment]
> I'm unclear if the `rustc` API offers a clean way to override the behavior of the `--version` flag

Doesn't look like it - the version string is baked in when `librustc_interface` is compiled: https://github.com/rust-lang/rust/blob/9110d925d0b3e4842376e830f4404a28e1aa2658/compiler/rustc_interface/src/util.rs#L654-L657
